### DISCUSSION TITLE
Testing MultiApps

### DISCUSSION THREAD

**Discussion by rtaylo45** [idaholab/moose#18328] [Q&A General] [answered]
Hello,

So I have a problem where I would like to test one way MultiApp coupling with my main app and a sub app. My main app and the sub app are developed by two different teams and not all users will have access to the sub app. The sub app solves for a number of variables which are required for my main app to use during its solve, so during coupling the sub app is called first. I was wondering if I could use an existing exidus output file to pull the required variable into my main app, to take out the requirement of the sub app. Maybe the restart and recovery system can help me do this? I know i can just apply functions as my AuxVariables used for coupling, but I thought it might add some robustness to testing if i used an existing output file. I am not fully aware how variable transfer happens under the hood, so my idea about how to do this might not add anything to testing. 

If anyone has ideas on how to do testing with MultiApps, I am all ears. 

Thanks,
Zack

**Comment by GiudGiud** [comment, marked as answer]
Hi Zack

I see at least two options here:
1) If you are fine with not testing the coupling part, you can reload the quantities that your subapp was providing:
- auxiliary variables using either exodus restart (add `use_for_exodus_restart` to [Mesh] and `initial_from_file_var =` for each variable) OR checkpoint restart
- postprocessors using a the `default` field on Receivers
The exodus/checkpoint to use would be previously generated using your subapp. 

2) if you want to test the coupling part (the transfers) but cannot provide the real subapp, then do `1)` but in the subapp, so perform all the restarts in a new emptied out subapp (you can add `solve = false` to Problem and `kernel_coverage_check = false` to make it very bare). 

Best,
Guillaume
